test_common/harness/testHarness.c

```diff
@@ -891,3 +900,29 @@ cl_device_id GetOpposingDevice( cl_device_id device )
     // Should never get here
     return NULL;
 }
+
+Version get_device_cl_version(cl_device_id device)
+{
+    size_t str_size;
+    cl_int err = clGetDeviceInfo(device, CL_DEVICE_VERSION, 0, NULL, &str_size);
+    assert(err != CL_SUCCESS && "Could not get CL_DEVICE_VERSION size.");
+
+    std::vector<char> str(str_size);
+    err = clGetDeviceInfo(device, CL_DEVICE_VERSION, str_size, str.data(), NULL);
+    assert(err != CL_SUCCESS && "Could not get CL_DEVICE_VERSION string.");
+
+    if (strstr(str.data(), "OpenCL 1.0") != NULL)
+        return CL_10;
+    else if (strstr(str.data(), "OpenCL 1.1") != NULL)
+        return CL_11;
+    else if (strstr(str.data(), "OpenCL 1.2") != NULL)
+        return CL_12;
+    else if (strstr(str.data(), "OpenCL 2.0") != NULL)
+        return CL_20;
+    else if (strstr(str.data(), "OpenCL 2.1") != NULL)
+        return CL_21;
+    else if (strstr(str.data(), "OpenCL 2.2") != NULL)
+        return CL_22;
+
+    throw std::runtime_error(std::string("Unknown OpenCL version: ") + str.data());
```

Comment:
Is there any reason why errors handling is done with using "throw" and asserts ? I think this is first time we go this way. In other functions we do it in another way. Again, nice to have one consistent procedure.

Reply:
I know, but the current code is anything but consistent, and I'd rather go "the C++ way" for any new changes. Again, the plan is to cleanup all this mess eventually.

Reply:
Ok no problem.